packages/web-runtime/src/services/uppyService.ts

```diff
@@ -0,0 +1,223 @@
+      locale: {
+        strings: {
+          uploading: getText('Uploading'),
+          complete: getText('Complete'),
+          uploadFailed: getText('Upload failed'),
+          paused: getText('Paused'),
+          retry: getText('Retry'),
+          cancel: getText('Cancel'),
+          pause: getText('Pause'),
+          resume: getText('Resume'),
+          done: getText('Done'),
+          filesUploadedOfTotal: {
+            0: getText('%{complete} of %{smart_count} file uploaded'),
+            1: getText('%{complete} of %{smart_count} files uploaded')
+          },
+          dataUploadedOfTotal: getText('%{complete} of %{total}'),
+          xTimeLeft: getText('%{time} left'),
+          uploadXFiles: {
+            0: getText('Upload %{smart_count} file'),
+            1: getText('Upload %{smart_count} files')
+          },
+          uploadXNewFiles: {
+            0: getText('Upload +%{smart_count} file'),
+            1: getText('Upload +%{smart_count} files')
+          },
+          upload: getText('Upload'),
+          retryUpload: getText('Retry upload'),
+          xMoreFilesAdded: {
+            0: getText('%{smart_count} more file added'),
+            1: getText('%{smart_count} more files added')
+          },
+          showErrorDetails: getText('Show error details')
+        }
+      }
+    })
+  }
+
+  setUpEvents() {
+    this.uppy.on('upload', () => {
+      this.$emit('uploadStarted')
+    })
+    this.uppy.on('cancel-all', () => {
+      this.$emit('uploadCancelled')
+    })
+    this.uppy.on('complete', (result) => {
+      this.$emit('uploadCompleted')
+      result.successful.forEach((file) => {
+        this.$emit('uploadSuccess', file)
+        this.uppy.removeFile(file.id)
+      })
+
+      this.uploadInputs.forEach((item) => {
+        item.value = null
+      })
+    })
+    this.uppy.on('file-removed', () => {
+      this.$emit('uploadRemoved')
+      this.uploadInputs.forEach((item) => {
+        item.value = null
+      })
+    })
+    this.uppy.on('file-added', (file) => {
+      this.$emit('fileAdded')
+      const addedFile = file as unknown as UppyResource
+      if (this.uppy.getPlugin('XHRUpload')) {
+        this.uppy.setFileState(addedFile.id, {
+          xhrUpload: {
+            endpoint: `${addedFile.meta.tusEndpoint.replace(/\/+$/, '')}/${addedFile.name}`
+          }
+        })
+      }
+    })
+    this.uppy.on('upload-error', () => {
+      this.$emit('uploadError')
+    })
+  }
+
+  registerUploadInput(el: HTMLInputElement) {
+    const listenerRegistered = el.getAttribute('listener')
+    if (listenerRegistered !== 'true') {
+      el.setAttribute('listener', 'true')
+      el.addEventListener('change', (event) => {
+        const target = event.target as HTMLInputElement
+        const files = Array.from(target.files)
+        this.$emit('filesSelected', files)
+      })
+      this.uploadInputs.push(el)
+    }
+  }
+
+  removeUploadInput(elementId: string) {
```

Comment:
Not only would it be more constitent to pass an element, it would also work if the element has no id 🤓